src/main/topLevelPages/myDashboard/newVirtualContributorWizard/useNewVirtualContributorWizard.tsx

```diff
@@ -148,57 +164,134 @@ const useNewVirtualContributorWizard = (): useNewVirtualContributorWizardProvide
+  };
+
+  const calloutId = useRef<string>();
+  const [createPost] = useCreatePostFromContributeTabMutation({
+    update: (cache, { data }) => {
+      if (!calloutId.current || !data) {
+        return;
+      }
+      const { createContributionOnCallout } = data;
+      const calloutRefId = cache.identify({
+        id: calloutId.current,
+      });
+
+      if (!calloutRefId) {
+        return;
+      }
 
-    const { data: newSpace } = await CreateNewSpace({
+      cache.modify({
+        id: calloutRefId,
+        fields: {
+          posts(existingPosts = []) {
+            const newPostRef = cache.writeFragment({
+              data: createContributionOnCallout.post,
+              fragment: PostCardFragmentDoc,
+              fragmentName: 'PostCard',
+            });
+            return [...existingPosts, newPostRef];
+          },
+        },
+      });
+    },
+  });
+
+  const onCreatePost = async (post: PostValues, calloutId: string) => {
+    await createPost({
       variables: {
-        hostId: user?.user.id,
-        spaceData: {
-          nameID: makeUniqueName(generateNameId(user?.user.profile.displayName!)),
-          profileData: {
-            displayName: generateSpaceName(user?.user.profile.displayName!),
+        postData: {
+          calloutID: calloutId!,
+          post: {
+            profileData: {
+              displayName: post.title,
+              description: post.description,
+            },
+            type: CalloutType.Post,
           },
-          collaborationData: {},
         },
-        licensePlanId: plans[0].id,
       },
     });
-    setSpaceId(newSpace?.createAccount.spaceID);
-
-    return newSpace;
   };
 
-  const handleSetupVirtualContributor = async (values: VirtualContributorFromProps) => {
-    setStep('create_VC');
-    if (mySpaceId && myAccountId) {
-      await handleCreateVirtualContributor(values, myAccountId, mySpaceId);
-    } else {
-      const newSpace = await handleCreateSpace();
-      await handleCreateVirtualContributor(values, newSpace?.createAccount.id!, newSpace?.createAccount.spaceID!);
+  const handleAddContent = async (posts: PostsFormValues) => {
+    setStep('vc_setup');
+
+    // create collection of posts
+    const callout = await handleCreateCallout(calloutDetails);
+    calloutId.current = callout?.id;
+
+    // add posts to collection
+    if (callout?.id && posts.posts.length > 0) {
+      posts.posts.forEach(async post => {
+        console.log(post);
+        await onCreatePost(post, callout?.id);
+      });
     }
-    await setStep('add_knowledge');
-  };
 
-  const handleAddContent = async () => {
-    console.log('Not implemented');
+    // create VC
+    if (virtualContributorInput && myAccountId && spaceId) {
+      handleCreateVirtualContributor(virtualContributorInput, myAccountId, spaceId);
+      addVCCreationCache(virtualContributorInput.name);
+      const { data } = await getNewSpaceUrl({ variables: { spaceId: spaceId! } });
```

Comment:
Can we get the URL with the response of space creation instead of making another call here?

Reply:
No. The mutation returns only id and spaceId

Reply:
@bobbykolev we have to keep this in mind yeah, `createAccount` is returning an `Account` which has a `spaceID` field that will be gone soon. We'll have to call createSpace and there it will return the url, for now I think this code is okay